### ISSUE TITLE
Unevaluated function

### ISSUE THREAD

**Issue by virgil-serbanuta** [kframework/kore#2725]
The predicate for one of my proofs contains an unevaluated function, and I think it should be easy to evaluate it. I'm using commit 252ff7bfeac9895d8511a4af6b7d33f4902ad749 , see below for how to reproduce. FWIW, this works at commit 1724c5c23afea515f4c58328d1f42abc5de6e15c (i.e. the function is evaluated, and, if the entire semantics is used, the proof works).

To be specific, the `countMapElementProperty` call below is not evaluated. The predicate contains this:
```
  {
    false
  #Equals
    applyMapElementProperty ( ?Key , ?Value )
  }
#And
  {
    true
  #Equals
    countMapElementProperty ( ?Key , ?Value ) +Int mapCount ( ?M ) >Int 0
  }
```
and my semantics contains this equation:
```
  rule countMapElementProperty(Key:KItem, Value:KItem)
      => 0
    requires notBool applyMapElementProperty(Key, Value)
```

By looking at the debug info for this equation, the side condition does not seem to contain anything related to `applyMapElementProperty` - the side condition is either `top` or it's a `\equals(false, in_keys(?Key, ?M))` (as the "Assumed true condition").

To reproduce:
------------------

Note that this is a minimized example which means, e.g., that `applyMapElementProperty` is tagged with `no-evaluators`, which is not the case in the full semantics. I could post an example with the entire semantics if needed.

[bug-files.zip](https://github.com/kframework/kore/files/6751231/bug-files.zip) - ~75 lines of code.

```
$  kompile tmp.k --backend haskell && kprove tmp-proof.k  --haskell-backend-command "kore-repl --repl-script kast.kscript"

step 2
select 2
konfig
```

**Comment by ana-pantilie** [comment]
As @JKTKops pointed out, this might be a duplicate of https://github.com/kframework/kore/issues/2785.

**Comment by ana-pantilie** [comment]
This should be fixed on ed422630f9383b2774f1ec8ea04eee86ea0afe1e. Please let me know if you still encounter issues.
